core/services/relay/evm/types/gen/main.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"bytes"
+	"go/format"
+	"os"
+	"text/template"
+)
+
+func main() {
+	genInts()
+	genBytes()
+}
+
+func genBytes() {
+	byteTypes := [33]ByteType{}
+	for i := 1; i < 33; i++ {
+		byteTypes[i-1].Size = i
+	}
+	runTemplate("bytes", bytesTemplate, "byte_types_gen.go", byteTypes)
+}
+
+func genInts() {
+	var intTypes []*IntType
+
+	// 8, 16, 32, and 64 bits have their own type in go that is used by abi.
+	// The test use *big.Int
+	for i := 24; i <= 256; i += 8 {
+		if i == 32 || i == 64 {
+			continue
+		}
+
+		signed := &IntType{Size: i, Signed: false}
+		unsigned := &IntType{Prefix: "u", Size: i}
+		intTypes = append(intTypes, signed, unsigned)
+	}
+	runTemplate("ints", intsTemplate, "int_types_gen.go", intTypes)
+}
+
+func runTemplate(name, rawTemplate, outputFile string, input any) {
+	t, err := template.New(name).Parse(rawTemplate)
+	if err != nil {
+		panic(err)
+	}
+
+	br := bytes.Buffer{}
+	if err = t.Execute(&br, input); err != nil {
+		panic(err)
+	}
+
+	res, err := format.Source([]byte(br.Bytes()))
+	if err != nil {
+		panic(err)
+	}
+
+	if err = os.WriteFile(outputFile, res, 0600); err != nil {
+		panic(err)
+	}
+}
+
+type IntType struct {
+	Prefix string
+	Size   int
+	Signed bool
+}
+
+type ByteType struct {
+	Size int
+}
+
+const bytesTemplate = `
+package types
+
+import "reflect"
+
+{{ range . }}
+type bytes{{.Size}} [{{.Size}}]byte
+func init() {
+	typeMap["bytes{{.Size}}"] = &AbiEncodingType {
+		Native: reflect.TypeOf([{{.Size}}]byte{}),
+		Checked: reflect.TypeOf(bytes{{.Size}}{}),
+	}
+}
+
+{{ end }}
+`
+
+const intsTemplate = `
```

Comment:
Maybe these are large enough to define in separate files and pull in with `package embed`?
I think IDEs handle them better that way too.

Reply:
Not a blocker though.